src/System.Collections/tests/Generic/Dictionary/Dictionary.Generic.Tests.netcoreapp.cs

```diff
@@ -55,5 +55,28 @@ public void GetValueOrDefault_KeyDoesntExist_ReturnsDefaultValue()
             Assert.Equal(default(TValue), dictionary.GetValueOrDefault(key));
             Assert.Equal(defaultValue, dictionary.GetValueOrDefault(key, defaultValue));
         }
+
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
+        public void Dictionary_Generic_TryAdd(int count)
+        {
+            var dictionary = (Dictionary<TKey, TValue>)GenericIDictionaryFactory(count);
+
+            TKey key = CreateTKey(seed: count);
+            TValue value = dictionary.GetValueOrDefault(key);
+            int originalCount = dictionary.Count;
+            bool alreadyContainsKey = dictionary.ContainsKey(key);
```

Comment:
In what situation will this already contain the key? And isn't / shouldn't this be deterministic?  Seems like the test should either assert that it contains the key or assert that it doesn't, based on the test's construction, rather than having to account for the possibility that it may or may not.

Even if you can't predict whether GenericIDictionaryFactory(count) will include the relevant key, we can still set up the test to ensure the key isn't there, e.g.
```C#
 var dictionary = (Dictionary<TKey, TValue>)GenericIDictionaryFactory(count); 
TKey key = CreateTKey(seed: count);
dictionary.Remove(key);
int originalCount = dictionary.Count;

TValue value = CreateTValue(seed: count);
Assert.True(dictionary.TryAdd(key, value));
Assert.Equal(originalCount + 1, dictionary.Count);
Assert.Equal(dictionary[key], value);

Assert.False(dictionary.TryAdd(key, default(TValue)));
Assert.Equal(originalCount + 1, dictionary.Count);
Assert.Equal(dictionary[key], value);
```